src/Microsoft.Tye.Extensions/Dapr/DaprExtension.cs

```diff
@@ -147,14 +169,14 @@ public override Task ProcessAsync(ExtensionContext context, ExtensionConfigurati
                         deployment.Annotations.Add("dapr.io/id", project.Name);
                         deployment.Annotations.Add("dapr.io/port", (httpBinding.Port ?? 80).ToString(CultureInfo.InvariantCulture));
 
-                        if (config.Data.TryGetValue("config", out var daprConfig) && daprConfig is object)
+                        if (config.Data.TryGetValue("config", out var obj) && obj?.ToString() is string daprConfig)
                         {
-                            deployment.Annotations.TryAdd("dapr.io/config", daprConfig!.ToString() ?? string.Empty);
+                            deployment.Annotations.TryAdd("dapr.io/config", daprConfig);
                         }
 
-                        if (config.Data.TryGetValue("log-level", out var logLevel) && logLevel is object)
+                        if (config.Data.TryGetValue("log-level", out obj) && obj?.ToString() is string logLevel)
```

Comment:
simplified this and made more resiliant. If `obj` is null or tostrings to null we will just skip.